Install the Logos Basecamp app (#262)
status-im/ift-docs: docs/core/journeys/install-the-logos-basecamp-app.md

---
title: Install the Logos Basecamp app
doc_type: procedure
product: core
topics: core
steps_layout: flat
authors: iurimatias, cheny0
owner: logos
doc_version: 1
slug: install-the-logos-basecamp-app
---

# Install the Logos Basecamp app

#### Get Logos Basecamp running on your desktop.

> [!NOTE]
>
> - **Permissions**: No special permissions required.
> - **Product**: Logos Basecamp.

Logos Basecamp is the desktop shell for the Logos platform. You can discover, install, and run Logos modules and apps using its graphical interface as an alternative to the command line.

You can install Logos Basecamp in two ways:

| Method | When to use it | Tooling required |
|:--|:--|:--|
| [Prebuilt release (AppImage or DMG)](#install-from-a-prebuilt-release) | End users | None |

> | [Build from source with Nix](#build-and-run-logos-basecamp-from-source) | Contributors, custom builds, unsupported platforms | Nix with flakes enabled |

Reviewer: I would add somewhere that `/etc/nix/config` needs to be changed to have `experimental-features = nix-command flakes` as this is not trivial to understand from this text and hard to find for anyone not familiar with `nix`(many people)

Before you start, make sure you have the following:

- Internet access.

> - A supported OS: Linux x86_64 or aarch64 (tested on Ubuntu 22.04+), or macOS x86_64 or aarch64 (recent versions).

Author: I don't see a macOS x86_64 dmg on the release page. Should macOS x86_64 be listed as a supported OS?

- 4 GB RAM minimum (8 GB recommended) and ~2 GB free disk space.
- For the source build only: [Nix](https://github.com/NixOS/nix-installer) installed with flakes enabled.

> [!NOTE]
>
> Internet access is required to download the binary or clone the repository, but not to launch Logos Basecamp afterward. Logos Basecamp itself opens no inbound ports. 

## Install from a prebuilt release

1. Go to the [latest release page](https://github.com/logos-co/logos-basecamp/releases/latest) and download the artifact for your OS:

>     - Linux: the `AppImage` files

Reviewer: I actually got into problems with `AppImage` on Ubuntu 24.04 with `No suitable fusermount binary found` and then `libEGL.so.1 / libOpenGL.so.0 missing`

what helped was to install `sudo apt-get install -y fuse libegl1 libopengl0`

so I am not sure of this claim (cc @iurimatias )

Reviewer 2: Added your solution in a new troubleshooting section; take a look

    - macOS: the `.dmg` files

1. Depending on your OS, install and launch the app as follows:

- On macOS, drag the `.dmg` file into `/Applications`. Then launch the app from `/Applications`.
- On Linux, grant execute permission to the downloaded AppImage and launch it:

    ```bash
    chmod +x logos-basecamp-*.AppImage
    ./logos-basecamp-x86_64.AppImage  # or logos-basecamp-aarch64.AppImage
    ```

## Build and run Logos Basecamp from source

1. Clone the repository and enter it:

    ```bash
    git clone https://github.com/logos-co/logos-basecamp.git
    cd logos-basecamp
    ```

1. Build the app with the Nix flake:

    ```bash
    nix build '.#app'
    ```

1. Run the resulting binary:

    ```bash

>     ./result/bin/logos-basecamp

Reviewer: this is correct build output 

```suggestion
    ./result/bin/LogosBasecamp
```

---
title: Install the Logos Basecamp app
doc_type: procedure
product: core
topics: core
steps_layout: flat
authors: iurimatias, cheny0
owner: logos
doc_version: 1
slug: install-the-logos-basecamp-app
---

# Install the Logos Basecamp app

#### Get Logos Basecamp running on your desktop.

> [!NOTE]
>
> - **Permissions**: No special permissions required.
> - **Product**: Logos Basecamp.

Logos Basecamp is the desktop shell for the Logos platform. You can discover, install, and run Logos modules and apps using its graphical interface as an alternative to the command line.

You can install Logos Basecamp in two ways:

| Method | When to use it | Tooling required |
|:--|:--|:--|
| [Prebuilt release (AppImage or DMG)](#install-from-a-prebuilt-release) | End users | None |
| [Build from source with Nix](#build-and-run-logos-basecamp-from-source) | Contributors, custom builds, unsupported platforms | Nix with flakes enabled |

> [!NOTE]
>

> > To enable flakes in nix, add `experimental-features = nix-command flakes` to `/etc/nix/config`.

Reviewer: **Wrong Nix config path.** The Nix configuration file is `/etc/nix/nix.conf` — `/etc/nix/config` does not exist, so this edit silently does nothing. Verified on a fresh install using the `nix-installer` linked just above: the active config lives at `/etc/nix/nix.conf` (Determinate Nix splits user overrides into a `!include nix.custom.conf`). Note the same installer already enables flakes by default, but if the manual step is kept the path must be correct.

```suggestion
> To enable flakes in nix, add `experimental-features = nix-command flakes` to `/etc/nix/nix.conf`.
```

Before you start, make sure you have the following:

- Internet access.
- A supported OS: Linux x86_64 or aarch64 (tested on Ubuntu 22.04+), or macOS x86_64 or aarch64 (recent versions).
- 4 GB RAM minimum (8 GB recommended) and ~2 GB free disk space.
- For the source build only: [Nix](https://github.com/NixOS/nix-installer) installed with flakes enabled.

> [!NOTE]
>
> Internet access is required to download the binary or clone the repository, but not to launch Logos Basecamp afterward. Logos Basecamp itself opens no inbound ports. 

## Install from a prebuilt release

1. Go to the [latest release page](https://github.com/logos-co/logos-basecamp/releases/latest) and download the artifact for your OS:
    - Linux: the `AppImage` files
    - macOS: the `.dmg` files

1. Depending on your OS, install and launch the app as follows:

- On macOS, drag the `.dmg` file into `/Applications`. Then launch the app from `/Applications`.
- On Linux, grant execute permission to the downloaded AppImage and launch it:

    ```bash

>     chmod +x logos-basecamp-*.AppImage
>     ./logos-basecamp-x86_64.AppImage  # or logos-basecamp-aarch64.AppImage

Reviewer: **These filenames don't match the actual release artifacts**, so following these two lines verbatim fails. The downloaded asset is named `LogosBasecamp-Desktop-<version>-<hash>-<arch>.AppImage` (current latest release `0.1.2`: `LogosBasecamp-Desktop-v0.1.2-2576ef-x86_64.AppImage`). Reproduced on a clean machine:

```
$ chmod +x logos-basecamp-*.AppImage
chmod: cannot access 'logos-basecamp-*.AppImage': No such file or directory
$ ./logos-basecamp-x86_64.AppImage
bash: ./logos-basecamp-x86_64.AppImage: No such file or directory
```

A version-agnostic glob matches the real artifact and launches the app (tested — it boots and loads modules):

```suggestion
    chmod +x LogosBasecamp-Desktop-*.AppImage
    ./LogosBasecamp-Desktop-*-x86_64.AppImage  # or the *-aarch64 build
```
